src/Tools/dotnet-monitor/Egress/AzureBlob/AzureBlobEgressProvider.cs

```diff
@@ -143,6 +166,19 @@ public AzureBlobEgressProvider(ILogger<AzureBlobEgressProvider> logger)
             }
         }
 
+        public void AddConfiguredMetadata(AzureBlobEgressProviderOptions options, EgressArtifactSettings artifactSettings)
+        {
+            if (options.IncludeHostNameAsMetadata)
+            {
+                artifactSettings.Metadata.Add(HostNameMetadataKey, Dns.GetHostName());
```

Comment:
I'm wary of using the current hostname of dotnet-monitor and reporting that as the hostname of the target process. You should probably get this from the environment block of the target process; the variable name is HOSTNAME if I recall correctly.